src/roles/common/tasks/check-predeploy-prereq.yml

```diff
@@ -20,8 +20,19 @@
 
 - block:
 
-  - block:
+  - name: Check that image files are not missing
+    set_fact:
+      is_file_missing: "{{ image_path | default ('MISSING') == 'MISSING' or vrs_package_path | default ('MISSING')  == 'MISSING' or dkms_package_path | default ('MISSING') == 'MISSING' }}"
+  
+  - name: Delete the md5 files
+    file:
+      path: "{{ inventory_dir }}/deployment.md5.yml"
+      state: absent
+    when: is_file_missing 
+    delegate_to: localhost
 
+  - block:
+    
     - name: Check that image files were found
       assert:
         that: "image_path | default('MISSING') != 'MISSING' and image_file_name | default('MISSING') != 'MISSING'"
```

Comment:
Please use:

that: not is_file_missing

as the condition here.  This removes the duplication of the checking code and ensures that all 3 of these tasks are always exactly in sync.